test/system/src/test/java/io/pravega/test/system/ReadWriteAndAutoScaleWithFailoverTest.java

```diff
@@ -0,0 +1,468 @@
+            for (int i = 0; i < INITIAL_NUM_WRITERS; i++) {
+                log.info("Starting writer{}", i);
+                final EventStreamWriter<Long> writer = clientFactory.createEventWriter(STREAM_NAME,
+                        new JavaSerializer<Long>(),
+                        EventWriterConfig.builder().retryAttempts(10).build());
+                writerList.add(writer);
+            }
+
+            log.info("Creating Reader group: {}, with readergroup manager using scope: {}", readerGroupName, scope);
+            readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                    Collections.singleton(STREAM_NAME));
+            log.info("Reader group name: {}, Reader group scope: {}, Online readers: {}",
+                    readerGroupManager.getReaderGroup(readerGroupName).getGroupName(), readerGroupManager
+                            .getReaderGroup(readerGroupName).getScope(), readerGroupManager
+                            .getReaderGroup(readerGroupName).getOnlineReaders());
+            log.info("Creating {} readers", NUM_READERS);
+            List<EventStreamReader<Long>> readerList = new ArrayList<>(NUM_READERS);
+            log.info("Scope that is seen by readers {}", scope);
+            for (int i = 0; i < NUM_READERS; i++) {
+                log.info("Starting reader: {}, with id: {}", i, readerName + i);
+                final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                        readerGroupName,
+                        new JavaSerializer<Long>(),
+                        ReaderConfig.builder().build());
+                readerList.add(reader);
+            }
+
+            // start writing asynchronously
+            List<CompletableFuture<Void>> writerFutureList = writerList.stream().map(writer ->
+                    startWriting(writer)).collect(Collectors.toList());
+
+            //start reading asynchronously
+            List<CompletableFuture<Void>> readerFutureList = readerList.stream().map(reader ->
+                    startReading(reader))
+                    .collect(Collectors.toList());
+
+            //run the failover test before scaling
+            performFailoverTest();
+
+            //bring the instances back to 3 before performing failover during scaling
+            controllerInstance.scaleService(3, true);
+            segmentStoreInstance.scaleService(3, true);
+            Thread.sleep(ZK_DEFAULT_SESSION_TIMEOUT);
+
+            //increase the number of writers to trigger scale
+            List<EventStreamWriter<Long>>  newlyAddedWriterList = new ArrayList<>(ADD_NUM_WRITERS);
+            log.info("Writers writing in the scope {}", scope);
+            for (int i = 0; i < ADD_NUM_WRITERS; i++) {
+                log.info("Starting writer{}", i);
+                final EventStreamWriter<Long> writer1 = clientFactory.createEventWriter(STREAM_NAME,
```

Comment:
Please do not use number suffixes in variables like `writer1`, it doesn't say much about it. Use for example, `tmpWriter`. Another thing is that you don't really need a new variable in every iteration of the loop, since you are simply adding it to a lost. Declare this writer variable outside the loop.